packages/flutter/test/material/user_accounts_drawer_header_test.dart

```diff
@@ -32,11 +60,17 @@ Future<void> pumpTestWidget(
         child: Material(
           child: Center(
             child: UserAccountsDrawerHeader(
-              onDetailsPressed: withOnDetailsPressedHandler ? () { } : null,
-              currentAccountPicture: const ExcludeSemantics(
-                child: CircleAvatar(
-                  key: avatarA,
-                  child: Text('A'),
+              onDetailsPressed: widget.withOnDetailsPressedHandler ? () { } : null,
+              currentAccountPicture: ExcludeSemantics(
+                child: GestureDetector(
```

Comment:
Since all you're trying to do is have `setState` called, I think it would be more clear to wrap your test widget in a [StatefulBuilder](https://docs.flutter.io/flutter/widgets/StatefulBuilder-class.html), store the setState method and then call it directly from your test.

See [this example](https://github.com/flutter/flutter/blob/0fb84e96c79df2fe7efeaeb06fbd1c2b53f4a84b/packages/flutter/test/material/will_pop_test.dart#L277), which I think is similar enough to what you want to do.

This will also make it so that you don't have to refactor `TestWidget` into a `StatefulWidget`.

Reply:
done